### DISCUSSION TITLE
HalfLife2 / Source Engine Build

### DISCUSSION THREAD

**Discussion by ferrellsl** [Joshua-Riek/ubuntu-rockchip#74] [General]
I was able to create a successful build of the source-engine (https://github.com/nillerusr/source-engine) with the following workflow.

1. Install the prerequisites as identified at: https://github.com/nillerusr/source-engine/wiki/Source-Engine-(EN)
2. Clone the repository: git clone https://github.com/nillerusr/source-engine --recursive --depth 1
3. Move to the source-engine folder: cd source-engine
4.  Create a folder on your Pi 5 Desktop called HL2
5. Start the WAF configuration tool: python3 waf configure -T release --64bits --prefix=~/Desktop/HL2
6. If the tool identifies any un-met dependencies that you've missed, install them and re-run the WAF configuration tool until they're all met.
7. Run the WAF compilation tool: python3 waf
8. Install the binaries: sudo python3 waf install
9. You should now have a hl2_launcher inside your HL2 folder.  
10.  Copy your Half-Life 2 steam folders over to your Pi 5 HL2 folder.  You will find them on your PC at:  x:\SteamLibrary\steamapps\common\Half-Life 2.
11. You will need the following folders and their contents:  hl2, bin, steam_input, and platform.  You can delete any DLL and EXE files from your Pi 5 after copying the folders over as they're for an x86/64 system.

You should now be able to run the hl2_launcher and start the game.

And some screenshots

![image](https://user-images.githubusercontent.com/5926287/225420459-a91f40b0-6374-409d-b7fe-d29e875762a8.png)
![image](https://user-images.githubusercontent.com/5926287/225420497-c0be3e7c-a30c-4ea1-a6b6-e8a3b7b5d754.png)
![image](https://user-images.githubusercontent.com/5926287/225420544-31f19429-08ab-4e70-b8df-e19839a9a4da.png)

**Comment by ferrellsl** [comment]
And here are some shots of in-game play:

![Screenshot from 2023-03-15 13-27-08](https://user-images.githubusercontent.com/5926287/225434613-2c697f9b-508b-46dd-a10a-296dffd2b8ec.png)
![Screenshot from 2023-03-15 13-27-50](https://user-images.githubusercontent.com/5926287/225434651-bb99273b-307e-4df8-ac87-1a1e9f0eb82b.png)
![Screenshot from 2023-03-15 13-28-13](https://user-images.githubusercontent.com/5926287/225434761-1361e3d8-9ac6-4cb5-a67f-900bb68ffe0a.png)

**Comment by ferrellsl** [comment]
![Screenshot from 2023-03-15 16-17-19](https://user-images.githubusercontent.com/5926287/225465164-96d78d78-9555-4b73-9bd8-fc709328ad1c.png)

![Screenshot from 2023-03-15 16-17-37](https://user-images.githubusercontent.com/5926287/225465218-843094ad-1975-4d5a-abfc-28113fd9db7b.png)

**Comment by ferrellsl** [comment]
I'm getting between 35 and 45 fps at 1080p as shown in the upper right corner of this screenshot. Sorry the text is so small.

![Screenshot from 2023-03-15 17-05-23](https://user-images.githubusercontent.com/5926287/225474100-a5ba2c3c-6563-4a70-acc2-18c067d5c326.png)

**Comment by canteroman** [comment]
how??? 

[ 135/2202] Compiling bitmap/colorconversion.cpp
[ 138/2202] Compiling bitmap/ImageByteSwap.cpp
../bitmap/colorconversion.cpp:29:10: fatal error: stb_dxt.h: No existe el archivo o el directorio
   29 | #include "stb_dxt.h"
      |          ^~~~~~~~~~~
compilation terminated.

**Reply by shipa-2** [reply]
skill issue

compiled full on Freescale i.mx6q
